docs/content/docs/deployment/config.md

```diff
@@ -27,16 +27,100 @@ under the License.
+Users can configure the values corresponding to the Config Type in the following format:
+
+{{< config_file >}}
+
+Additionally, users can configure the value for all Config Types as strings by simply enclosing the original value in single quotes or double quotes.
+
+
+### Migrate from flink-conf.yaml to config.yaml
+#### Behavior Changes
+`config.yaml` strictly follows the YAML 1.2 syntax and is compatible with `flink-conf.yaml` in most cases, except for the following behavior changes:
+
+- Null value: 
+  - `flink-conf.yaml`: Only supports leaving the value blank.
+  - `config.yaml`: Supports leaving it blank, or explicitly set it to null, Null, NULL, or `~`.
+
+- Comment: 
+  - `flink-conf.yaml`: Everything after the first occurrence of `#` in each line is considered a comment.
+  - `config.yaml`: When there is at least one space between the `#` and the preceding content, or when the `#` is at the beginning of a line, the subsequent content is considered a comment.
+
+- Special character escaping in strings: 
+  - `flink-conf.yaml`: Only needs to escape elements in Lists and Maps.
+    - List elements containing a semicolon ";" require escaping.
+    - Map elements containing a comma "," or colon ":" require escaping.
+  - `config.yaml`: Requires escaping special characters as defined in the YAML 1.2 specification; see the definition of [special characters](https://yaml.org/spec/1.2.2/#53-indicator-characters).
+
+- Duplicated keys: 
+  - `flink-conf.yaml`: Allows duplicated keys and takes the last key-value pair for the corresponding key that appears in the file.
+  - `config.yaml`: Does not allow duplicated keys, and an error will be reported when loading the configuration.
+
+- Handling of invalid configuration: 
+  - `flink-conf.yaml`: Invalid key-value pairs will be ignored.
+  - `config.yaml`: An error will be reported when loading the configuration.
+
+#### Migration Tool
+To facilitate user migration, Flink provides a configuration file migration script that can automate the migration process. The usage is as follows:
+
+- Place the old configuration file `flink-conf.yaml` in the `conf/` directory.
+- Execute the following command in the `$FLINK_HOME/` directory:
+````migrate-tool.sh
+bin/migrate-config-file.sh
+````
+After running the command above, the migration script will automatically read the old configuration file `flink-conf.yaml` from the `conf/` directory and output the migrated results to the new configuration file `config.yaml` in the `conf/` directory. Note that due to historical reasons, all values in flink-conf.yaml will be recognized as String type, so the values in the generated config.yaml file will also be of String type, which means some values will be enclosed in quotes. However, Flink will convert them to the actual types defined in the configuration during subsequent configuration parsing.
```

Comment:
> due to historical reasons

due to the limitation of the legacy configuration parser

> defined in the configuration

defined using `ConfigOption`